tests/framework/extensions/projects/projectmap.go

```diff
@@ -0,0 +1,15 @@
+package projects
+
+import (
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+)
+
+func ProjectConfig(clusterID string) (projectConfig *management.Project) {
```

Comment:
It seems like a constructor template. Do we need this to be an extension? If yes, we have a few examples `workload/templates` for more template examples. Maybe changing a few things might make this a solid template constructor.

Reply:
Public functions need their own godoc comment explanation. Could we add an explanation?

Reply:
yes, I think this would be a good case for extensions because projects can be created not just in rbac test but any test for example, charts, or even provisioning etc. By having this added as an extension we can avoid adding the same config in multiple places.

Sure, I can take a look at the examples and update accordingly.